src/main/java/de/tum/in/www1/artemis/repository/BonusRepository.java

```diff
@@ -0,0 +1,62 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.Bonus;
+
+/**
+ * Spring Data JPA for the BonusSource entity
+ */
+@Repository
+public interface BonusRepository extends JpaRepository<Bonus, Long> {
+
+    /**
+     * Find a bonus source by its source grading scale id
+     *
+     * @param sourceGradindScaleId the source grading scale id
+     * @return an Optional with the bonus source if such scale exists and an empty Optional otherwise
+     */
+    Optional<Bonus> findBySourceGradingScaleId(@Param("sourceGradindScaleId") Long sourceGradindScaleId);
```

Comment:
❌

```suggestion
    Optional<Bonus> findBySourceGradingScaleId(@Param("sourceGradingScaleId") Long sourceGradindScaleId);
```

Also I do not think you need the `@Param` here

Reply:
I consider this change mandatory.

Reply:
Removed the method altogether since it is not currently used and it is easy to re-add if it is needed.